clients/cobol-lsp-vscode-extension/src/services/ControlFlowService.ts

```diff
@@ -168,11 +177,25 @@ export class ControlFlowAnalysisService implements AnalysisServiceDelegate {
     mainChannel?: vscode.OutputChannel,
     logChannel?: vscode.LogOutputChannel,
   ) {
-    const service = new ControlFlowAnalysisService(mainChannel, logChannel);
     return (result: ApiResult) => {
-      service.handleControlFlowAst(result).catch(() => {});
+      ControlFlowAnalysisService.getInstance(mainChannel, logChannel)
+        .handleControlFlowAst(result)
+        .catch(() => {});
     };
   }
+
+  private static getInstance(
+    mainChannel?: vscode.OutputChannel,
+    logChannel?: vscode.LogOutputChannel,
+  ): ControlFlowAnalysisService {
+    if (!ControlFlowAnalysisService.instance) {
+      ControlFlowAnalysisService.instance = new ControlFlowAnalysisService(
```

Comment:
Please do not re-introduce the singleton, it unnecessarily complicates testing. Speaking of testing - I don't see any.